### PR TITLE
First draft of annotation filtering

### PR DESCRIPTION
**Pull request by matentzn**
I tried this on all ontologies; seems to work. I will leave the rest to you @dosumis

### CODE REVIEW

**Comment by matentzn** [comment]
(i added Pato as an example)

**Comment by dosumis** [comment]
Still seeing a lot of cruft in Protege.  e.g. PATO import has:

image

Do I need to run update imports locally to test?

**Comment by matentzn** [comment]
I  have now updated pato.. No it was a major headache with make.. GR. Check out pato import now!

File: src/ontology/cl-edit.owl [hunk 1]
```diff
@@ -2403,7 +2403,6 @@ Declaration(Class(obo:CP_0000039))
 Declaration(Class(obo:CP_0000040))
 Declaration(Class(obo:CP_0000043))
 Declaration(Class(obo:D96882F1-8709-49AB-BCA9-772A67EA6C33))
```
Comment (cmungall) [inline src/ontology/cl-edit.owl:2405]: this is pretty wacky (not in your PR, I know)

**Review by cmungall — APPROVED** [review]
The allowed list looks good. I don't see the logic for getting subAPs of synonym_type and subset_type

**Comment by matentzn** [comment]
not there yet, will do it tomorrow.

**Comment by matentzn** [comment]
I am doing this now:

```
remove -T allowed_annotations.txt --select "self descendants" --select complement --select annotation-properties 
```

which is a bit of a trick, assuming that we generally want to include child APs. If we didnt, we would have to create special handling for subsets - like this it is generic (in the case of PATO, does exactly what what we want). The downside is that this could pull in a few APs we didnt want - but these would not show up in the Protege list, since they are children.

**Review by dosumis — APPROVED** [review]

**Comment by dosumis** [comment]
Updating other imports before I merge.

**Comment by dosumis** [comment]
(I guess this might cause problems with import updates in branches + merging ...)

**Comment by matentzn** [comment]
No worries at all. Its just one, and that is mine (I can fix the conflicts quickly)

**Pull request merged**